application-tour-ui/src/main/resources/TourCode/AjaxActions.xml

```diff
@@ -0,0 +1,536 @@
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators> ,|</separators>
+        <size>1</size>
+        <sort>none</sort>
+        <unmodifiable>0</unmodifiable>
+        <validationMessage/>
+        <validationRegExp/>
+        <values>top|left|right|bottom</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </placement>
+      <targetPage>
+        <customDisplay/>
+        <disabled>0</disabled>
+        <name>targetPage</name>
+        <number>7</number>
+        <picker>0</picker>
+        <prettyName>Target Page (optional)</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <validationMessage/>
+        <validationRegExp/>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </targetPage>
+      <title>
+        <customDisplay/>
+        <disabled>0</disabled>
+        <name>title</name>
+        <number>2</number>
+        <picker>0</picker>
+        <prettyName>Title</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <validationMessage/>
+        <validationRegExp/>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </title>
+    </class>
+    <name>TourCode.AjaxActions</name>
+    <number>6</number>
+    <className>TourCode.StepClass</className>
+    <guid>863176a8-a939-4a95-9635-8dbd979ed4aa</guid>
+    <property>
+      <backdrop>0</backdrop>
+    </property>
+    <property>
+      <content/>
+    </property>
+    <property>
+      <element>xcxc</element>
+    </property>
+    <property>
+      <order>1</order>
+    </property>
+    <property>
+      <placement/>
+    </property>
+    <property>
+      <title>jxcx</title>
+    </property>
+  </object>
+  <content>{{velocity}}
+#**
+* Ajax Actions
+*
+* Author: Mohamed BOUSSAA &lt;mohamed.boussa@xwiki.com&gt; 
+*#
+
+#set($action = $request.getParameter("action"))
+#if ("$!action" != '')
+#if ($action == "editStep" &amp;&amp; "$!request.getParameter('BSTDocRef')" != '')
+  #set($docRef = $request.getParameter("BSTDocRef"))
+  #if ("$!docRef" != '')
+     #set($bootstrapTourDoc = $xwiki.getDocument($docRef))
+     #if ($bootstrapTourDoc)
+        ## Save step
+        #set($title = $request.getParameter("title"))
+        #set($path = $request.getParameter("path"))
+        #set($element = $request.getParameter("element"))
+        #set($backdrop = $request.getParameter("backdrop"))
+        #set($content = $request.getParameter("content"))
+        #set($placement = $request.getParameter("placement"))
+        #set($targetPage = $request.getParameter("targetPage"))
+        #set($nbSteps = $request.getParameter("nbSteps"))
+        #set($stepOrder = $request.getParameter("stepOrder"))
+        #set($isNewStep = true)
+        #if ("$!stepOrder" != '') ## Edit
+            #set($stepObj = $bootstrapTourDoc.getObject("TourCode.StepClass", "order", "$stepOrder"))
```

Comment:
I think you should not rely on the "order" field to identify an object, but on the real object number instead.